src/wp-includes/option.php

```diff
@@ -776,6 +776,24 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	if ( has_filter( "pre_option_{$option}" ) ) {
```

Comment:
What about the `pre_option` filter?

Reply:
@spacedmonkey I agree that in principle there could be the same bug due to usage of `pre_option` too, but I intentionally did not consider the general `pre_option` filter, because it may have wider reaching implications, so I think unhooking it temporarily is too risky.

There's a good chance that a site that uses that filter has some special mechanisms in place for retrieving and storing options that shouldn't be unhooked.

That is different with the `pre_option_{$option}` filter, which is clearly about specific options.